launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -433,6 +439,36 @@ QProcessEnvironment MinecraftInstance::createEnvironment()
     return env;
 }
 
+QProcessEnvironment MinecraftInstance::createLaunchEnvironment()
+{
+    // prepare the process environment
+    QProcessEnvironment env = createEnvironment();
+
+#ifdef Q_OS_LINUX
+    if (settings()->get("EnableMangoHud").toBool())
+    {
+        auto preload = env.value("LD_PRELOAD", "") + ":libMangoHud_dlsym.so:libMangoHud.so";
+        auto lib_path = env.value("LD_LIBRARY_PATH", "") +  ":/usr/local/$LIB/:/usr/local/$LIB/mangohud/:/usr/$LIB/mangohud/";
+
+        env.insert("LD_PRELOAD", preload);
+        env.insert("LD_LIBRARY_PATH", lib_path);
+        env.insert("MANGOHUD", "1");
+    }
+
+    if (settings()->get("UseDiscreteGpu").toBool())
+    {
+        // Open Source Drivers    
```

Comment:
```suggestion
        // Open Source Drivers
```
nit: trailing spaces